spec/slack/real_time/celluloid_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+
+RSpec.describe Slack::RealTime::Celluloid::Socket do
+  context 'with url' do
+    let(:url) { 'wss://echo.websocket.org/websocket/xyz' }
+    subject(:socket) { described_class.new(url, ping: 42) }
+    let(:driver) { WebSocket::Driver::Client }
+    let(:ws) { double(driver) }
+
+    describe '#initialize' do
+      it 'sets url' do
+        expect(socket.url).to eq url
+      end
+    end
+
+    describe '#connect!' do
+      before do
+        allow(ws).to receive(:on).with(:close)
+      end
+
+      xit 'connects' do
```

Comment:
Stub away the loopy parts. Like `EM.start do ...` can be `allow(EM).to receive(:start).and_yield`